generators/create-react-app/templates/circleci.yaml.ejs

```diff
@@ -81,6 +85,56 @@ jobs:
       - run: sentry-cli releases new <%= projectName %>-<%= packageName %>@${CIRCLE_SHA1}
       - run: sentry-cli releases files <%= projectName %>-<%= packageName %>@${CIRCLE_SHA1} upload-sourcemaps <%= packagePath %>/build
       - run: sentry-cli releases finalize <%= projectName %>-<%= packageName %>@${CIRCLE_SHA1}
+<% } %>
+
+<% if (deployment === 'kubernetes') { %>
+  <%= packageName %>-build:
+    executor: docker/docker
+    steps:
+      - setup_remote_docker
+      - checkout
+      - docker/build:
+          image: <%= projectName %>-<%= packageName %>
+          tag: << pipeline.git.branch >>
+          path: <%= packagePath %>
+          docker-context: <%= packagePath %>
+          extra_build_args: '--build-arg VERSION=$(git log --max-count 1 --pretty=format:%H "<%= packagePath %>")'
+  <%= packageName %>-build-and-push:
+    executor: docker/docker
+    steps:
+      - setup_remote_docker
+      - checkout
+      - docker/build:
+          registry: $DOCKER_REGISTRY
+          image: <%= projectName %>-<%= packageName %>
+          tag: << pipeline.git.branch >>
+          path: <%= packagePath %>
+          docker-context: <%= packagePath %>
+          extra_build_args: '--build-arg VERSION=$(git log --max-count 1 --pretty=format:%H "<%= packagePath %>")'
+      - docker/build:
+          registry: $DOCKER_REGISTRY
+          image: <%= projectName %>-<%= packageName %>
+          tag: << pipeline.git.branch >>-sentry
+          path: <%= packagePath %>
+          docker-context: <%= packagePath %>
+          extra_build_args: '--build-arg VERSION=$(git log --max-count 1 --pretty=format:%H "<%= packagePath %>") --target sentry'
+      - docker/check:
+          registry: $DOCKER_REGISTRY
+      - docker/push:
+          registry: $DOCKER_REGISTRY
+          image: <%= projectName %>-<%= packageName %>
+          tag: << pipeline.git.branch >>
+      - run:
+          name: Publish source maps to sentry
+          command: |
+            IMAGE="$DOCKER_REGISTRY/<%= projectName %>-<%= packageName %>:<< pipeline.git.branch >>-sentry"
+            ENV="--env SENTRY_AUTH_TOKEN --env SENTRY_ORG --env SENTRY_PROJECT=<%= projectName %>-<%= packageName %>"
+            VERSION=<%= projectName %>-<%= packageName %>@$(git log --max-count 1 --pretty=format:%H "<%= packagePath %>")
+
+            docker run --rm $ENV $IMAGE releases new $VERSION
+            docker run --rm $ENV $IMAGE releases files $VERSION upload-sourcemaps .
+            docker run --rm $ENV $IMAGE releases finalize $VERSION
```

Comment:
This will try to recreate a new release in case the CRA directory didn't change but I've tested it on a project and Sentry doesn't error in this case so I decided to leave the code like this to simplify the config instead of adding a test for an existing release for this version.